src/tracks/playabletrack/wavetrack/ui/ClipOverflowButtonHandle.cpp

```diff
@@ -0,0 +1,103 @@
+#include "ClipOverflowButtonHandle.h"
+#include "BasicUI.h"
+#include "ClipParameters.h"
+#include "HitTestResult.h"
+#include "ProjectAudioIO.h"
+#include "ProjectHistory.h"
+#include "ProjectWindows.h"
+#include "RefreshCode.h"
+#include "UndoManager.h"
+#include "WaveClip.h"
+#include "WaveClipUtilities.h"
+#include "WaveTrack.h"
+#include "ZoomInfo.h"
+
+namespace
+{
+constexpr auto buttonWidth = 24;
+
+wxRect GetButtonRectangle(const wxRect& affordanceRect)
+{
+   const auto x = affordanceRect.x + affordanceRect.width - buttonWidth;
+   return { x, affordanceRect.y, buttonWidth, affordanceRect.height };
+}
+} // namespace
+
+ClipOverflowButtonHandle::ClipOverflowButtonHandle(
+   const std::shared_ptr<WaveTrack>& track,
+   const std::shared_ptr<WaveTrack::Interval>& clip)
+    : mTrack { track }
+    , mClip { clip }
+{
+}
+
+bool ClipOverflowButtonHandle::HitTest(
+   const ClipTimes& clip, const ZoomInfo& viewInfo, const wxRect& viewRect,
+   const wxPoint& point)
+{
+   const auto clipRect = ClipParameters::GetClipRect(clip, viewInfo, viewRect);
+   return GetButtonRectangle(clipRect).Contains(point);
+}
+
+wxRect
+ClipOverflowButtonHandle::DrawOnClip(wxDC& dc, const wxRect& affordanceRect)
+{
+   const auto rect = GetButtonRectangle(affordanceRect);
+   // Invisible pen
+   constexpr auto penWidth = 0;
+   wxDCPenChanger pen { dc, { wxPenInfo { wxColour {}, penWidth } } };
+   wxDCBrushChanger brush { dc, *wxBLACK_BRUSH };
+   constexpr auto numDots = 3;
+   constexpr auto radius = 1;
+   constexpr auto spacing = 4;
+   constexpr auto totalWidth = numDots * 2 * radius + 2 * spacing;
+   const auto dotY = rect.y + (rect.height - radius) / 2;
+   for (auto i = 0; i < numDots; ++i)
+   {
+      const auto dotX =
+         rect.x + (rect.width - totalWidth) / 2 + i * (2 * radius + spacing);
+      dc.DrawCircle(dotX, dotY, radius);
+   }
+
+   return { affordanceRect.x, affordanceRect.y,
+            affordanceRect.width - rect.width, affordanceRect.height };
+}
+
+UIHandle::Result ClipOverflowButtonHandle::Click(
+   const TrackPanelMouseEvent& event, AudacityProject* pProject)
+{
+   BasicUI::CallAfter(
+      [=] { ShowClipPitchAndSpeedDialog(*pProject, *mTrack, *mClip); });
+   return 0;
+}
+
+UIHandle::Result ClipOverflowButtonHandle::Drag(
+   const TrackPanelMouseEvent& event, AudacityProject* pProject)
+{
+   return 0;
+}
+
+HitTestPreview ClipOverflowButtonHandle::Preview(
+   const TrackPanelMouseState& state, AudacityProject* pProject)
+{
+   static auto cursor = std::make_unique<wxCursor>(wxCURSOR_HAND);
+   return { XO("Open clip Pitch and Speed menu"), cursor.get(),
+            XO("Pitch and Speed...") };
```

Comment:
It seems like there have been some wires crossed here - the overflow button should summon the clip context menu. Clicking on the pitch/speed numbers should summon the dialog.